OpenRA.Mods.Common/Util.cs

```diff
@@ -175,5 +175,23 @@ public static int RandomDelay(World world, int[] range)
 
 			return world.SharedRandom.Next(range[0], range[1]);
 		}
+
+		static WDist allMinimumRequiredScanRadius;
+		static WDist blockerMinimumRequiredScanRadius;
+		static Ruleset lastRules = null;
+
+		public static WDist MinimumRequiredScanRadius(Ruleset rules, ActorInfo ai, bool onlyBlockers = false)
+		{
+			if (lastRules != rules)
```

Comment:
Putting this in a trait would require a world trait; otherwise, the calculations will still be evaluated for each actor with one of the using traits and each actor with one of the using traits would be storing the same calculation result. Perhaps, `ActorMap` would be the best place to store the calculations; the calculations could be lazy but would not need to check which ruleset is in use.